### PR TITLE
#4077 historical sync stall after restart in era 0 on pre-1.5 chain

### LINKED ISSUE #4077: Investigate stuck historical sync on large network test
Investigate stuck historical sync issue reported here: https://casper-labs-team.slack.com/archives/C01ULKL8G8J/p1687350784393109

### PR DESCRIPTION
**Pull request by EdHastingsCasperAssociation**
see: 
- https://github.com/casper-network/casper-node/issues/4077
- https://github.com/casper-network/casper-node/pull/4116
- https://github.com/casper-network/casper-node/pull/4081

this is an alternative attempt at it

### CODE REVIEW

**Comment by EdHastingsCasperAssociation** [comment]
bors try

File: node/src/reactor/main_reactor/keep_up.rs [hunk 1]
```diff
@@ -601,74 +599,19 @@ impl MainReactor {
...
-                            }
-                        };
-                        Ok(Some(SyncBackInstruction::Sync {
-                            parent_hash: *parent_hash,
-                            era_id,
-                        }))
-                    }
-                    Err(err) => Err(err.to_string()),
-                }
+                self.validator_matrix
```
Comment (alsrdn) [inline node/src/reactor/main_reactor/keep_up.rs:609]: nit: it would be good to preserve the previous comment here

**Review by alsrdn — APPROVED** [review]

File: node/src/reactor/main_reactor/keep_up.rs (hunk 1 above)
Comment (EdHastingsCasperAssociation) [reply node/src/reactor/main_reactor/keep_up.rs:609]: it is retained, it was extracted into the synched function below along w/ the logic

**Review by fizyk20 — APPROVED** [review]

**Pull request merged**